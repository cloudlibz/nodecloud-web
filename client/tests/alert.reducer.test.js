import * as reducer from "../src/_reducers/alert.reducer";
import * as actions from "../src/_actions/alert.actions";

describe("reducer", () => {
  describe("ALERT SUCCESS", () => {
    test("sets alert to success", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = actions.alertActions.success;
      const nextState = reducer.alert(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("ALERT ERROR", () => {
    test("sets alert to error", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = actions.alertActions.error;
      const nextState = reducer.alert(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("ALERT CLEAR", () => {
    test("sets alert to clear", function() {
      const previousState = {
        otherProp: "test-123"
      };
      const action = actions.alertActions.clear;
      const nextState = reducer.alert(previousState, action);
      expect(nextState).toMatchSnapshot();
    });
  });
});
