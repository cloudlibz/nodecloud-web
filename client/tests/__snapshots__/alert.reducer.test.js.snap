// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer ALERT CLEAR sets alert to clear 1`] = `
Object {
  "otherProp": "test-123",
}
`;

exports[`reducer ALERT ERROR sets alert to error 1`] = `
Object {
  "otherProp": "test-123",
}
`;

exports[`reducer ALERT SUCCESS sets alert to success 1`] = `
Object {
  "otherProp": "test-123",
}
`;
